src/databases/block_database.cpp

```diff
@@ -174,12 +174,13 @@ bool block_database::top(size_t& out_height, bool candidate) const
 block_result block_database::get(size_t height, bool candidate) const
 {
     auto& manager = candidate ? candidate_index_ : confirmed_index_;
-    const auto link = height < manager.count() ? read_index(height, manager) :
-        hash_table_.not_found;
+    const auto element = height < manager.count()
+        ? hash_table_.find(read_index(height, manager))
+        : hash_table_.terminator();
 
     return
     {
-        hash_table_.find(link),
+        element,
```

Comment:
This introduces a copy. Could change element to non-const and apply a `std::move(element)`, though an explicit move is not necessary as efficient, depends on the compiler.